### PR TITLE
add examples for /api/pattern docs

### PR DESCRIPTION
**Pull request by iamsdas**
Signed-off-by: Suryashankar Das <suryashankardas.2002@gmail.com>

**Description**
This PR adds examples for `/api/pattern` and `/api/pattern/{id}` POST and GET APIs to the the docs.

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by iamsdas** [comment]
@sayantan1413 could you review it?

**Comment by l5io** [comment]
🚀 Preview for commit 42a735318335ce52514ac216e6c019f87405a618 at: https://6109674786f40451ebabd146--meshery-docs-preview.netlify.app

**Review by sayantan1413 — COMMENTED** [review]
Great work @iamsdas 🎉.  @leecalcote can you verify it if this was the change that you were asking for?

File: docs/_data/swagger.yml [hunk 1]
```diff
@@ -209,6 +255,24 @@ paths:
                 type: string
                 x-go-name: UserID
             x-go-package: github.com/layer5io/meshery/models
+          examples:
```
Comment (leecalcote) [inline docs/_data/swagger.yml:258]: Cool. These are looking good. Do you suppose that we could have an example of a pattern that was uploaded to Meshery from GitHub? // @Chadha93

**Comment by piyushsingariya** [comment]
Sorry, but I would like to point out that this file (docs/_data/swagger.yml) is automatically overwritten by github-actions/swagger-commands, so added custom-changes will be lost when this file will be updated.

**Comment by iamsdas** [comment]
Is there any way to locally generate the `swagger.yml` from the annotations? I'm getting a very different file when generating it with `go-swagger`. I believe we can add the example directly to the swagger annotations in the `mesheryPatternsResponseWrapper` wrapper.

**Comment by piyushsingariya** [comment]
> Is there any way to locally generate the `swagger.yml` from the annotations? I'm getting a very different file when generating it with `go-swagger`. I believe we can add the example directly to the swagger annotations in the `mesheryPatternsResponseWrapper` wrapper.

Yes that's possible, would you like to try that?

**Comment by iamsdas** [comment]
@piyushsingariya yes I am trying it out, but am currently stuck because the locally generated `swagger.yml` is very different from the pre-existing `swagger.yml`

**Comment by piyushsingariya** [comment]
I'm not sure what command are you using, you can use the ones pre-written in Makefile of Meshery folder.

File: docs/_data/swagger.yml (hunk 1 above)
Comment (Chadha93) [reply docs/_data/swagger.yml:258]: name: IstioSM
version: 1.0.1
services:
  istio:
    type: IstioMesh
    namespace: istio-system
    settings:
      version: 1.9.1
    traits:
      mTLS:
        policy: mutual
        namespaces:
          - istio-test
      automaticSidecarInjection:
        namespaces:
          - default
          - istio-test
​
  grafana:
    type: GrafanaIstioAddon
    namespace: istio-system
    dependsOn:
      - istio
      - prometheus
​
  prometheus:
    type: PrometheusIstioAddon
    namespace: istio-system
    dependsOn:
      - istio
      
  prometheus2:
    type: PrometheusIstioAddon
    namespace: istio-system
    dependsOn:
      - istio

**Comment by l5io** [comment]
🚀 Preview for commit 1211fe62b8f8c0bdc2f8efe0715f84b58fb41e24 at: https://611e0e72f0ec08f55dc22c33--meshery-docs-preview.netlify.app

**Comment by iamsdas** [comment]
The docs are finally getting generated from the swagger annotations, but its kind of messy right now and wont work if there is more than one level of nesting of examples. And also I am having some issues getting some objects/arrays examples to get generated properly. I have only added the example for GET request of pattern API. Does anyone have any better idea to do this?

**Comment by l5io** [comment]
🚀 Preview for commit 3f9fbcc139d50254e444b9a082820313037555fc at: https://611e11d566f277bf89e886f4--meshery-docs-preview.netlify.app

File: models/meshery_pattern.go [hunk 2]
```diff
@@ -10,17 +10,21 @@ import (
...
+	// example: string
+	Name string `json:"name,omitempty"`
+	// example: string
 	PatternFile string `json:"pattern_file"`
 	// Meshery doesn't have the user id fields
 	// but the remote provider is allowed to provide one
+	// example: string
 	UserID *string `json:"user_id" gorm:"-"`
-
+	// example: {"additionalProp1":{},"additionalProp2":{},"additionalProp3":{}}
```
Comment (utkarsh-pro) [inline models/meshery_pattern.go:23]:
```suggestion
	// example: { "host": "github.com", "path": "service-mesh-patterns/service-mesh-patterns", "type": "remote" }
```

**Pull request closed without merging**